test/langtools/jdk/jshell/LocalExecutionControlExceptionTest.java

```diff
@@ -0,0 +1,154 @@
+                Assertions.assertEquals(0, out.size());
+            }
+            jshell.eval("""
+                        <T extends Throwable> T t2(Throwable t) throws T {
+                            throw (T) t;
+                        }
+                        """);
+            jshell.eval("""
+                        void t(Throwable t) {
+                            throw t2(t);
+                        }
+                        """);
+            {
+                var events = jshell.eval("""
+                                         {
+                                             var t = new Thread(() -> t(new java.io.IOException()));
+                                             t.start();
+                                             t.join();
+                                         }
+                                         """);
+                Assertions.assertEquals(1, events.size());
+                Assertions.assertNull(events.get(0).exception());
+                Assertions.assertEquals(0, excs.size());
+                Assertions.assertEquals(0, out.size());
+            }
+            {
+                var events = jshell.eval("""
+                                         {
+                                             var t = new Thread(() -> t(new IllegalAccessException()));
+                                             t.start();
+                                             t.join();
+                                         }
+                                         """);
+                Assertions.assertEquals(1, events.size());
+                Assertions.assertNull(events.get(0).exception());
+                Assertions.assertEquals(0, excs.size());
+                Assertions.assertEquals(0, out.size());
+            }
+            Thread outsideOfJShell = new Thread(() -> {
+                t(new IOException());
+            });
+            outsideOfJShell.start();
+            outsideOfJShell.join();
+            Assertions.assertEquals(1, excs.size());
+        }
+    }
+
+    void t(Throwable t) {
+        throw t2(t);
+    }
+
+    <T extends Throwable> T t2(Throwable t) throws T {
+        throw (T) t;
+    }
+
+    public static final class NoUncaughtExceptionHandleInterceptor implements InvocationInterceptor {
+        public void interceptTestMethod(Invocation<Void> invocation,
+                                        ReflectiveInvocationContext<Method> invocationContext,
+                                        ExtensionContext extensionContext) throws Throwable {
+            Throwable[] exc = new Throwable[1];
+            //the tests normally run in a ThreadGroup which handles uncaught exception
+            //run in a ThreadGroup which does not handle the uncaught exceptions, and let them
+            //pass to the default uncaught handler for the test:
+            var thread = new Thread(Thread.currentThread().getThreadGroup().getParent(), "test-group") {
```

Comment:
Is there a reason we cannot use this for `outsideOfJShell` above so we don't need this aop thing?

Reply:
Sorry, I don't quite get the question.

IIRC, the purpose o this code is to avoid/opt-out from the auto-handling of "unhandled" exceptions.

I think the `outsideOfJShell` part above is there mostly to make sure this opt-out works correctly, and is not checking JShell as such.